### ISSUE TITLE
NuGet package does not indicate downstream dependencies

### ISSUE THREAD

**Issue by colinangusmackay** [plivo/plivo-dotnet#17]
The Plivo code relies on RestSharp. Packages.config states:

    <package id="RestSharp" version="104.4.0" targetFramework="net40" />

However the NuSpec file does not contain this dependency, so when including the Plivo NuGet package in my project it does not pull the downstream dependencies and as a result it fails to run.

NuSpec file needs something like the following, in order to correctly install in target projects:

    <package>
      <metadata>
      
        <dependencies>
          <group targetFramework="net40">
            <dependency id="RestSharp" version="104.4.0" />
          </group>
        </dependencies>
      </metadata>
      
    </package>

**Comment by EIrwin** [comment]
I added the snippet above to the NuSpec file.

**Comment by colinangusmackay** [comment]
I should also additionally say that the NuGet package should not contain the RestSharp DLL file. The nuspec file should be explicit (naming the exact DLLs that are to be included) rather than implicit (using wild cards that pick up other files that happen to the in the bin folder.)

**Comment by EIrwin** [comment]
@colinangusmackay are you referring to the following?

    <file src="bin\Release\*.dll" target="lib\net40" />

If so, if I am understanding this correctly, you want this to read the following

    <file src="bin\Release\RestSharp.dll" target="lib\net40" />

Let me know if that is correct.

**Comment by colinangusmackay** [comment]
Yes, No

RestSharp should NOT be included in the NuGet package. Only the fact that the package is dependent (as in my original post) should be in the NuGet package (via the NuSpec file). That's the way NuGet is supposed to work.

The DLLs that are actually in the NuGet package should be explicitly named, that's the Plivo.dll file in this case.

Incidentally, while I'm here. The NuGet package that was released this morning ([v1.3.6](https://www.nuget.org/packages/Plivo/1.3.6)) appears to be broken and effectively uninstalled Plivo. I send a message via NuGet.org to the project owner to this effect too.

**Comment by EIrwin** [comment]
@colinangusmackay I modified the NuSpec file to use the following:

    <files>
       <file src="bin\Release\Plivo.dll" target="lib\net40" />
    </files>
    <dependencies>
       <group targetFramework="net40">
         <dependency id="RestSharp" version="104.4.0" />
       </group>
    </dependencies>

I apologize for the confusion, but I think that is what you were asking for. Let me know if I have misunderstood.

**Comment by EIrwin** [comment]
@colinangusmackay Has there been any update regarding this issue?

**Comment by EIrwin** [comment]
@tsudot @colinangusmackay I haven't heard anything regarding the status of this.

**Comment by tsudot** [comment]
@EIrwin This is already included in this pull request - https://github.com/plivo/plivo-dotnet/pull/16

**Comment by colinangusmackay** [comment]
I've not heard anything official from anyone. I did briefly have an email exchange with someone at Plivo, but that was initiated because one of their sales/marketing people contacted me to see how my trial was going. They said they'd look into it, but I never heard back from them again.

In then end I just pulled the source directly into my solution and fixed it there. I will go back to the NuGet package when able to do so.
